### PR TITLE
[Sprint 2]Logic update profile

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Link remind: https://dev.framgia.com/redmine/issues/112981
Task: Update user information
Affect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfilePresenter.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        if (TextUtils.isEmpty(currentPassword)) {
+            isValidate = false;
+            mViewModel.onInputCurrentPasswordError();
+        }
+        if (TextUtils.equals(currentPassword, confirmNewPassword)) {
+            isValidate = true;
+        } else {
+            mViewModel.onCheckConfirmPasswordError();
+        }
+
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfilePresenter.java]: remove empty line

File: app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfilePresenter.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public void onStop() {
+    }
+
+    @Override
+    public void updateProfile(String newPassword, String chatWorkId, String description,
+            String currentPassword) {
+            //TODO: Handle update information
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfilePresenter.java]: check if User does not edit any field. We will not call to sever in here

File: app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfileViewModel.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    private UpdateProfileContract.Presenter mPresenter;
+    private Navigator mNavigator;
+    private User mUser;
+    private String mEmail;
+    private String mUsername;
+    private String mNewPassword;
+    private String mConfirmNewPassword;
+    private String mChatworkId;
+    private String mDescription;
+    private String mCurrentPassword;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfileViewModel.java]: remove!

File: app/src/main/res/layout/fragment_profile_update.xml [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+                    android:layout_alignBottom="@+id/user_profile_photo"
+                    android:layout_alignEnd="@+id/user_profile_photo"
+                    android:layout_alignRight="@+id/user_profile_photo"
+                    android:src="@drawable/ic_camera"
+                    />
+
+            </RelativeLayout>
+
+            <LinearLayout
+                android:id="@+id/linearLayout3"
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_profile_update.xml]: change name id to ``container_...``

File: app/src/main/res/layout/fragment_profile_update.xml [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center"
+                    android:layout_marginTop="@dimen/dp_5"
+                    android:background="@color/color_white"
+                    android:orientation="horizontal"
+                    android:padding="@dimen/dp_5"
+                    >
+
+                    <ImageView
+                        android:id="@+id/icon_email"
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_profile_update.xml]: check name convention
https://github.com/daolq3012/coding-standards/blob/master/vn/android/standard.md

File: app/src/main/res/layout/fragment_profile_update.xml [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center"
+                    android:layout_marginTop="@dimen/dp_5"
+                    android:background="@color/color_white"
+                    android:orientation="horizontal"
+                    android:padding="@dimen/dp_5"
+                    >
+
+                    <ImageView
+                        android:id="@+id/icon_current_password"
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_profile_update.xml]: https://github.com/daolq3012/coding-standards/blob/master/vn/android/standard.md

File: app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfileViewModel.java (hunk 3 above)
Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfileViewModel.java]: Chỗ này em follow theo trên web đó anh, phải nhập mật khẩu hiện tại mới cho update information anh ơi

Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/profilepage/updateprofile/UpdateProfileViewModel.java]: à OK!

**Pull request merged**